clean up recipe and convert to noarch:python

synapticarbors opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

synapticarbors committed:
clean up recipe and convert to noarch:python

synapticarbors commented:
@conda-forge-admin please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

Commit:
MNT: Re-rendered with conda-build 25.1.1, conda-smithy 3.45.4, and conda-forge-pinning 2025.02.10.18.05.55

synapticarbors commented:
@jkochNU @petergthatsme -- We were seeing some issues in downstream packages on conda-forge so I took a look at this recipe and noticed that there doesn't appear to be any compiled extensions. I therefore converted it to a `noarch: python` package and cleaned up some dependencies. Importantly I added `setuptools-scm` so that the version reported by the package would be correct instead of `0.0.0`.

I'd appreciate if you could take a look at this and merge it if looks good to you.

synapticarbors commented:
One thing that I was unsure of was the python specific version constraints on `scipy` for mac upstream. Can you explain why these are necessary? I didn't try to constrain them here since I couldn't find any mention in the upstream repo of the reason behind this.

kochjens commented:
> One thing that I was unsure of was the python specific version constraints on `scipy` for mac upstream. Can you explain why these are necessary? I didn't try to constrain them here since I couldn't find any mention in the upstream repo of the reason behind this.

We were facing issues with scipy versions from two sides, historically: there were qutip versions that were incompatible with certain scipy versions (without that being caught by qutip); there were certain scipy versions that caused issue with ARM architecture on Macs. Would it simplify things if we just suppressed certain scipy choices on all platforms?
 
 Thank you for all your help here, much appreciated!

Harrinive commented on recipe/meta.yaml:
We need to include `{% set python_min = "3.9" %}` at the beginning.

kochjens replied:
That line is now included in the submitted meta.yml

kochjens committed:
Update meta.yaml with python_min info
